tools/autograd/templates/Functions.cpp

```diff
@@ -665,6 +665,20 @@ Tensor var_backward(Tensor grad, const Tensor & self, IntArrayRef dim, bool unbi
   return (2.0 / (_safe_size(self.sizes(), dim) - unbiased)) * grad * (self - self.mean(dim, true));
 }
 
+Tensor var_std_mean_backward(const variable_list& grads, const Tensor & self, const Tensor & r1, const Tensor & r2, IntArrayRef dim, bool unbiased, bool keepdim, bool is_std) {
+  AT_CHECK(grads.size() > 0 && grads.size() <= 2, "grads size should be in interval [0,2]");
+  std::vector<Tensor> results;
+  if (grads[0].defined()) {
+    Tensor grad = is_std ? static_cast<Tensor>(grads[0]) / (r1 * 2) : static_cast<Tensor>(grads[0]);
+    results.push_back(var_backward(grad, self, dim, unbiased, keepdim));
+  }
+  if (grads[1].defined()) {
+    Tensor grad = static_cast<Tensor>(grads[1]);
+    results.push_back(sum_backward(grad, self.sizes(), dim, keepdim) / _safe_size(self.sizes(), dim));
+  }
+  return at::stack(results, 0);
```

Comment:
This doesn't seem correct, because the size of the backward depends on the number of grad_outputs?

Also, I can't even seem to take a backwards of these functions.  Consider:
```
>>> x=torch.randn(2,3, requires_grad=True)
>>> torch.std_mean(x)
(tensor(0.9186), tensor(-0.0833))
```
neither output requires grad.  So what are the autograd tests actually doing?

Reply:
Fixed

Reply:
this isn't exactly fixed, the grad_input size is still wrong (it should be the same as the input size), the autograd engine is just unexpanding it here: https://github.com/pytorch/pytorch/blob/87a69741934b5c41684027af7e481c0a0e13d69f/torch/csrc/autograd/engine.cpp#L373.  In other words, you are doing an extremely inefficient add.

Reply:
Fixed